File: src/types/brandVoice.ts
export interface StyleAttributes {
  tone: string;
  vocabulary_level: string;
  sentence_style: string;
  personality_traits: string[];
  signature_phrases: string[];
  avoid_phrases: string[];
}

export interface BrandVoiceProfile {
  id: string;
  name: string;
  description: string | null;
  style_attributes: StyleAttributes;
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

File: src-tauri/src/models/brand_voice.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BrandVoiceProfile {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub style_attributes: StyleAttributes,
    pub is_default: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StyleAttributes {
    pub tone: String,
    pub vocabulary_level: String,
    pub sentence_style: String,
    pub personality_traits: Vec<String>,
    pub signature_phrases: Vec<String>,
    pub avoid_phrases: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AnalyzeVoiceRequest {
    pub name: String,
    pub description: Option<String>,
    pub samples: Vec<String>,
}
